docs/design/downward_api_resources_limits_requests.md

```diff
@@ -0,0 +1,530 @@
+     // Required: Selects a field of the pod: only annotations, labels, name and  namespace are supported.
+     FieldRef ObjectFieldSelector `json:"fieldRef"`
+     // Required: Selects a field of the container: only resources limits and requests (cpu, memory) are supported.
+     ContainerFieldRef ObjectFieldSelector `json:"containerFieldRef,omitempty"`
+}
+
+
+// EnvVarSource represents a source for the value of an EnvVar.
+// Only one of its fields may be set.
+type EnvVarSource struct {
+   // Required: Selects a field of the container: only resources limits and requests (cpu, memory) are supported.
+   ContainerFieldRef *ObjectFieldSelector `json:"containerFieldRef,omitempty"`
+   // Selects a field of the pod; only name and namespace are supported.
+   FieldRef *ObjectFieldSelector `json:"fieldRef,omitempty"`
+   // Selects a key of a ConfigMap.
+   ConfigMapKeyRef *ConfigMapKeySelector `json:"configMapKeyRef,omitempty"`
+   // Selects a key of a secret in the pod's namespace.
+   SecretKeyRef *SecretKeySelector `json:"secretKeyRef,omitempty"`
+}
+
+// ObjectFieldSelector selects an APIVersioned field of an object.
+type ObjectFieldSelector struct {
+        // Required: Version of the schema the FieldPath is written in terms of.
+        // If no value is specified, it will be defaulted to the APIVersion of the
+        // enclosing object.
+        APIVersion string `json:"apiVersion"`
+        // Required: Path of the field to select in the specified API version
+        FieldPath string `json:"fieldPath"`
+}
+```
+
+#### Examples
+
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: dapi-test-pod
+spec:
+  containers:
+    - name: test-container
+      image: gcr.io/google_containers/busybox command: [ "/bin/sh","-c", "env" ]
+      resources:
+        requests:
+          memory: "64Mi"
+          cpu: "250m"
+        limits:
+          memory: "128Mi"
+          cpu: "500m"
+      env:
+        - name: CPU_LIMIT
+          valueFrom:
+            containerFieldRef:
+              fieldPath: resources.limits.cpu
+```
+
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: kubernetes-downwardapi-volume-example
+spec:
+  containers:
+    - name: client-container
+      image: gcr.io/google_containers/busybox command: ["sh", "-c", "while true; do if [[ -e /etc/labels ]]; then cat /etc/labels; fi; if [[ -e /etc/annotations ]]; then cat /etc/annotations; fi; sleep 5; done"]
+     resources:
+        requests:
+          memory: "64Mi"
+	  cpu: "250m"
+        limits:
+          memory: "128Mi"
+          cpu: "500m"
+      volumeMounts:
+        - name: podinfo
+          mountPath: /etc
+          readOnly: false
+  volumes:
+    - name: podinfo
+      downwardAPI:
+        items:
+          - path: "container_name/cpu_limit"
+            containerFieldRef:
+              fieldPath: resources.limits.cpu
+```
+
+### API with no selectors
+
+In this approach, users specify particular strings to retrieve resources
```

Comment:
@aveshagarwal We want to avoid creating new magic keys, so you can either just add a new option to pick from, or rewrite the third option.  Maybe it would be better to add a new one.